sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/aio/_patch.py

```diff
@@ -62,12 +62,30 @@ def __init__(
         ledger_certificate_path: Union[bytes, str, os.PathLike],
         **kwargs: Any,
     ) -> None:
+        # Remove some kwargs first so that there aren't unexpected kwargs passed to
+        # get_ledger_identity.
+        if isinstance(credential, ConfidentialLedgerCertificateCredential):
+            auth_policy = None
+        else:
+            credential_scopes = kwargs.pop("credential_scopes", ["https://confidential-ledger.azure.com/.default"])
+            auth_policy = kwargs.pop(
+                "authentication_policy",
+                # Don't call AsyncBearerTokenCredentialPolicy here as an event loop may not be
+                # running if a non-async authentication policy is passed.
+                None,
+            )
+
+            if auth_policy is None:
+                auth_policy = policies.AsyncBearerTokenCredentialPolicy(credential, *credential_scopes, **kwargs)
+
         if os.path.isfile(ledger_certificate_path) is False:
             # We'll need to fetch the TLS certificate.
             identity_service_client = ConfidentialLedgerCertificateClient(**kwargs)
 
             # Ledger URIs are of the form https://<ledger id>.confidential-ledger.azure.com.
             ledger_id = endpoint.replace("https://", "").split(".")[0]
+
+            # We use the sync client here because async __init__ is not allowed.
```

Comment:
This is....problematic.
If a customer passes in their own transport, it will all fall apart - which is something all our SDKs are expected to support (as well as being able to pass transport-specific sessions for connection pooling). They may also pass things like async raw request/response hooks which would also fail here.
Given that this is already in the SDK - we don't have to address it in this PR - but we should open an issue here... I can't recall if this was discussed prior to GA?

Reply:
I feel like I discussed it with someone, but I have no record of it, maybe it was on a call.

The gist of the problem is that we can't make an `async` call inside `__init__`, but we need a network call in the `__init__` for the certificate retrieval magic.

If the customer provides a transport, they likely intend to use it for normal client calls, not the magical one. Further, there's no chance any async hooks will work in the constructor anyway. 

Is it a reasonable solution to simply not pass the `kwargs` to any of the `ConfidentialLedgerCertificateClient`-related calls? This way they will always be synchronous. Of course, we will still be doing a magical sync call in an async client, but that seems to be the only way to achieve the certificate-fetching convenience inside the constructor.

Reply:
Not passing kwargs is not an option - if a customer needs to configure requests for their environment (e.g. routing through a proxy) then they will need to be configured here too.
We usually have other ways of approaching the "need to make a request in the constructor" issue - as this is not the only SDK where's it's come up.

Reply:
This also creates issues for dependencies - at it requires me to keep sync transport dependencies around (e.g. `requests`) for no apparent reason.

Reply:
Makes sense, I will open an issue for this!

Reply:
Created an issue here: https://github.com/Azure/azure-sdk-for-python/issues/31512